From e4e69d7a766681842094781772e2627307741a08 Mon Sep 17 00:00:00 2001
From: Hezi <hezi@marvell.com>
Date: Fri, 9 Oct 2015 16:40:08 +0300
Subject: [PATCH 048/538] ASoc: kirkwood: Set Initial I2S PLL Frequency

Details:
 Some Codecs rely on the I2S PLL MClk.
 The default I2S PLL MClk rate is 48MHz
 This is too high for some codecs.
 We set it to about 11.2MHz which is suitable
 for the 44.1KHz HiFi Audio rate.

Signed-off-by: Hezi <hezi@marvell.com>
Signed-off-by: Hezi Shahmoon <hezi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/23904
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Neta Zur Hershkovits <neta@marvell.com>
Change-Id: Iec6fd48c51a6adaf4aab6a619280e68e1591ed99
Reviewed-on: http://vgitil04.il.marvell.com:8080/26887
Reviewed-by: Lior Amsalem <alior@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27280
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Lior Amsalem <alior@marvell.com>
---
 sound/soc/kirkwood/kirkwood-i2s.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/kirkwood/kirkwood-i2s.c b/sound/soc/kirkwood/kirkwood-i2s.c
index f189141..9bbf613 100644
--- a/sound/soc/kirkwood/kirkwood-i2s.c
+++ b/sound/soc/kirkwood/kirkwood-i2s.c
@@ -679,6 +679,9 @@ static int kirkwood_i2s_dev_probe(struct platform_device *pdev)
 
 	if (of_device_is_compatible(np, "marvell,armada-380-audio")) {
 		err = armada_38x_i2s_init_quirk(pdev, priv, soc_dai);
+		/* Set initial pll frequency */
+		if (priv->pll_config)
+			armada_38x_set_pll(priv->pll_config, 44100);
 		if (err < 0)
 			return err;
 	}
-- 
1.9.1

